wordle-web/pages/wordleOfTheDay.vue

```diff
@@ -0,0 +1,169 @@
+
+      <v-btn @click="game.submitGuess(true, userName, calcSecond())" class="mb-5" color="primary"
+        >Guess!</v-btn
+      >
+
+      <v-btn
+        class="mb-5 ml-5"
+        color="primary"
+        variant="flat"
+        @click="router.push('/leaderboard')"
+        >Leaderboard</v-btn
+      >
+    </v-card>
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import { Game, GameState } from "../scripts/game";
+import Axios from "axios"; //npm install axios
+
+const router = useRouter();
+const showHelpDialog = ref(false);
+const userName: Ref<string> = inject("userName")! as Ref<string>;
+const game = reactive(new Game());
+game.startNewGameAPI();
+provide("GAME", game);
+const showUserNameDialog = ref(false);
+var startTime = new Date().getTime();
+
+//watch for the route to change to this page, then check if the user is a guest or not then prompt the user to enter a name
+
+onMounted(() => {
+  checkUserName();
+  window.addEventListener("keyup", onKeyup);
+});
+
+function onKeyup(event: KeyboardEvent) {
+  if (event.key === "Enter") {
+    game.submitGuess(true, userName.value);
+  } else if (event.key == "Backspace") {
+    game.removeLastLetter();
+  } else if (event.key.match(/[A-z]/) && event.key.length === 1) {
+    game.addLetter(event.key.toUpperCase());
+  }
+}
+function calcAttempts() {
+  var attempts = 0;
+  if (game.gameState == GameState.Won) {
+    attempts = game.guessIndex + 1;
+  } else {
+    attempts = game.guesses.length + 5;
+  }
+  return attempts;
+}
+function postScore(playerNameIn: string, attemptsIn: number, timeIn: number) {
+  console.log("score data: " + playerNameIn + " " + attemptsIn + " " + timeIn);
+  let postScoreUrl = "Player/UpdateScore";
+  Axios.post(postScoreUrl, {
+    playerName: playerNameIn,
+    attempts: attemptsIn,
+    time: timeIn,
+  }).then((response) => {
+    console.log("response from API post score " + response.data + " " + response.status);
+  });
+  
+}
+function calcSecond() {
+  var endTime = new Date().getTime();
+  var timeDiff = endTime - startTime;
+  return timeDiff / 1000;
+}
+function checkUserName() {
+  if (userName.value === "guest" || userName.value === "") {
+    showUserNameDialog.value = true;
+  }
+}
+watch(
+  () => game.gameState,
+  (value) => {
+    if (value == GameState.Won || value == GameState.Lost) {
+      if (userName.value === "guest" || userName.value === "") {
+        showUserNameDialog.value = true;
+        watch(
+          () => showUserNameDialog.value,
+          (value) => {
+            if (value == false) {
+              if(userName.value ===""){
+                userName.value = "guest";
+              }
+              postScore(userName.value, calcAttempts(), calcSecond());
+            }
+          }
+        );
+      } else {
+        postScore(userName.value, calcAttempts(), calcSecond());
+      }
+
+      //I know userName is showing an error but the api only gets the data when its set up like that
+    }
+  }
+);
```

Comment:
Having a little trouble following this here, there's a watcher defined inside the call back of another watcher here, which is something i've never seen before, and honestly not sure what kind of behavior this produces, but I imagine something unexpected and possibly related to your comment on line 165. 

Without knowing exactly what the intent was, below is my best attempt at a suggestion.

```suggestion
watch(
  () => game.gameState,
  (value) => {
    if (value === GameState.Playing && (!userName.value || userName.value === "guest")){
      showUserNameDialog.value = true;
      return;
    }
    postScore(userName.value ?? "guest", calcAttempts(), calcSecond());
    //I know userName is showing an error but the api only gets the data when its set up like that
  }
);
```

Reply:
the second watcher is watching for the username dialog to be closed after prompting the user to enter another username if their name is guest or blank